src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Abstractions/Mappers/AssemblyMapper.cs

```diff
@@ -7,15 +7,20 @@
 namespace Microsoft.DotNet.ApiCompatibility.Abstractions
 {
     /// <summary>
-    /// Object that represents a mapping between two <see cref="IAssemblySymbol"/> objects.
+    /// Object that represents a mapping between multiple <see cref="IAssemblySymbol"/> objects.
     /// This also holds a list of <see cref="NamespaceMapper"/> to represent the mapping of namespaces in between
     /// <see cref="ElementMapper{T}.Left"/> and <see cref="ElementMapper{T}.Right"/>.
     /// </summary>
-    public class AssemblyMapper : ElementMapper<IAssemblySymbol>
+    public class AssemblyMapper : ElementMapper<ElementContainer<IAssemblySymbol>>
```

Comment:
This means that the AssemblyMapper now isn't just a collection of `IAssemblySymbol` but a tuple of `IAssemblySymbol` and `MetadataInformation`. Ideally this would be flattened to just be `ElementMapper<IAssemblySymbol, MetadataInformation` but that's not easily possible with the current inheritance structure of the other mappers. So nesting is the current best trade-off.